### PR TITLE
Adding maximize flag to Adagrad

### PR DESCRIPTION
**Pull request by zaxtax**
This adds maximize to Adagrad (#68052) along with updates the respective tests.

### CODE REVIEW

File: torch/optim/adagrad.py [hunk 1]
```diff
@@ -230,34 +281,43 @@ def _single_tensor_adagrad(params: List[Tensor],
...
+            params,
+            grads,
+            state_sums,
+            state_steps,
+            lr=lr,
+            weight_decay=weight_decay,
+            lr_decay=lr_decay,
+            eps=eps,
+            has_sparse_grad=has_sparse_grad,
+            maximize=maximize,
```
Comment (albanD) [inline torch/optim/adagrad.py:319]: The grads were already flipped here no?

**Review by albanD — COMMENTED** [review]
Sounds  mostly good (if I didn't miss small change in the middle of the formatting).

File: torch/optim/adagrad.py (hunk 1 above)
Comment (zaxtax) [reply torch/optim/adagrad.py:319]: Oops yes!

**Review by albanD — APPROVED** [review]
Change looks good.
We need to make sure the CI properly runs and is green though!

**Comment by albanD** [comment]
This one is some CI infra failure, never seen it before. Rebasing on top of master should fix this.

**Pull request closed without merging**